app/scripts/components/common/catalog/prepare-datasets.ts

```diff
@@ -80,6 +80,15 @@ export function prepareDatasets (
       return a[sortField]?.localeCompare(b[sortField]);
     });
 
+  if (filterLayers && sortField) {
+    filtered = filtered.map((d) => ({
+      ...d,
+      layers:
+        isDatasetData(d) &&
+        d.layers.sort((a, b) => a[sortField]?.localeCompare(b[sortField]) || 0)
+    })) as DatasetData[];
+  }
```

Comment:
I think https://github.com/NASA-IMPACT/veda-ui/blob/03d0488ce02a0fc474a5deb16df93828056f6e6e/app/scripts/components/common/catalog/prepare-datasets.ts#L75-L81 should already work?

Reply:
The problem is that this will only work for the Data Catalog, where we show the datasets. But in the E&A Modal we show the layers within those datasets, so we have to go one level down (to the layers prop) to sort them.